### ISSUE TITLE
Accidental bypass for react/jsx-no-bind allowArrowFunctions

### ISSUE THREAD

**Issue by richardson-trevor** [yannickcr/eslint-plugin-react#1395]
We've recently started to get rid of our usage of arrow functions in component props for similar reasons to that mentioned [here](https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36). We went ahead and set: 
```JS
'react/jsx-no-bind': ['error', {
      allowArrowFunctions: false,
      allowBind: false,
}]
```
but realized we were still able to get away with using something like:
```JS
const someFunc = () => onClickHandler(userId)
<SomeComponent onClick={someFunc} />
```
So we didn't fix the problem, we just accidentally tricked it. Is this expected behavior or is this a bug?

**Comment by ljharb** [comment]
It's not realistic for a rule to always be exhaustive; there's often a way to trick them.

However, it seems reasonable that an arrow function in the same scope should be warned on.

**Comment by richardson-trevor** [comment]
Agreed. The primary concern was that for someone not familiar with the implications of anonymous functions in props, they might think the rules is just stylistic. I think checking current scope would reject the 99% case of extracting to a variable.
